src/net_processing.cpp

```diff
@@ -3439,6 +3439,7 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
             }
             if (!fRevertToInv && !vHeaders.empty()) {
                 if (vHeaders.size() == 1 && state.fPreferHeaderAndIDs) {
+                    assert(pBestIndex);
```

Comment:
This is a matter of taste, but I think that the more explicit `pBestIndex != nullptr` is more readable as to the intent.  I'm not sure if there's a style rule for how to test pointers against null which goes against that, though.